ang/civiawards/reviews-tab/directives/reviews-case-tab-content.directive.js

```diff
@@ -201,10 +222,40 @@
             responses.awardDetails.review_fields,
             responses.scoringFields
           );
+
+          return fetchOptionValues();
         })
         .finally(function () {
           $scope.isLoading = false;
         });
     }
+
+    /**
+     * Fetches Option Values for every scoring fields
+     *
+     * @returns {Promise} promise
+     */
+    function fetchOptionValues () {
+      var optionGroupIDs = [];
+      var optionGroupApiCalls = [];
+
+      _.each($scope.scoringFields, function (field) {
+        if (field.option_group_id) {
+          optionGroupIDs.push(field.option_group_id);
+          optionGroupApiCalls.push(['OptionValue', 'get', {
+            sequential: 1,
+            return: ['label', 'value'],
+            option_group_id: field.option_group_id
+          }]);
+        }
+      });
+
+      return crmApi(optionGroupApiCalls)
+        .then(function (responses) {
+          _.each(optionGroupIDs, function (optionGroupID, index) {
+            optionGroupValues[optionGroupID] = _.indexBy(responses[index].values, 'value');
+          });
+        });
+    }
```

Comment:
I think it will be better to group the option group api call in a single request:

```js
var optionGroupIds = _.chain($scope.scoringFields)
  .filter(function (field) { return !!field.option_group_id; })
  .map('option_group_id')
  .value();

return crmApi('OptionValue', 'get', {
  sequential: 1,
  return: ['label', 'option_group_id', 'value'],
  option_group_id: { IN: optionGroupIds }
})
  .then(function (response) {
    optionGroupValues = _.chain(optionGroupIds)
      .mapKeys()
      .mapValues(function (optionGroupId) {
        return _.chain(response.values)
          .filter({ option_group_id: optionGroupId })
          .indexBy('value')
          .value();
      });
  });
```

This is a bit verbose, but we can do some optimizations by moving things to different functions:

```js
function fetchOptionValues () {
  var optionGroupIds = getScoringFieldsGroupIds();

  return crmApi('OptionValue', 'get', {
    sequential: 1,
    return: ['label', 'option_group_id', 'value'],
    option_group_id: { IN: optionGroupIds }
  })
    .then(function (response) {
      optionGroupValues = getIndexedOptionGroupValues(optionGroupIds, response.values);
    });
}

function getScoringFieldsGroupIds () {
  return _.chain($scope.scoringFields)
    .filter(function (field) { return !!field.option_group_id; })
    .map('option_group_id')
    .value();
}

function getIndexedOptionGroupValues (optionGroupIds, optionGroupValues) {
  return _.chain(optionGroupIds)
    .mapKeys()
    .mapValues(function (optionGroupId) {
      return _.chain(optionGroupValues)
        .filter({ option_group_id: optionGroupId })
        .indexBy('value')
        .value();
    });
}
```

Reply:
As a general note: it's prefered to use map instead of pushing values into an array directly. There are some advantages to this approach, mainly that is functional (no side effects) and it's easy to move to a new function if we want to tidy up the code.